packages/manifold/contracts/gachaclaims/ERC1155GachaLazyClaim.sol

```diff
@@ -0,0 +1,250 @@
+      cost: claimParameters.cost,
+      erc20: claim.erc20
+    });
+    emit GachaClaimUpdated(creatorContractAddress, instanceId);
+  }
+
+  /**
+   * See {IERC1155GachaLazyClaim-getClaim}.
+   */
+  function getClaim(address creatorContractAddress, uint256 instanceId) public view override returns (Claim memory) {
+    return _getClaim(creatorContractAddress, instanceId);
+  }
+
+  /**
+   * See {IERC1155GachaLazyClaim-getClaimForToken}.
+   */
+  function getClaimForToken(
+    address creatorContractAddress,
+    uint256 tokenId
+  ) external view override returns (uint256 instanceId, Claim memory claim) {
+    instanceId = _tokenInstances[creatorContractAddress][tokenId];
+    claim = _getClaim(creatorContractAddress, instanceId);
+  }
+
+  function _getClaim(address creatorContractAddress, uint256 instanceId) private view returns (Claim storage claim) {
+    claim = _claims[creatorContractAddress][instanceId];
+    if (claim.storageProtocol == StorageProtocol.INVALID) revert IGachaLazyClaim.ClaimNotInitialized();
+  }
+
+  /**
+   * See {IGachaLazyClaim-mintReserve}.
+   */
+  function mintReserve(address creatorContractAddress, uint256 instanceId, uint32 mintCount) external payable override {
+    require(!Address.isContract(msg.sender), "Only EOAs can reserve a mint.");
+    _mintReserve(creatorContractAddress, instanceId, mintCount);
+  }
+
+  function _mintReserve(address creatorContractAddress, uint256 instanceId, uint32 mintCount) private {
+    Claim storage claim = _getClaim(creatorContractAddress, instanceId);
+    // Checks for reserving
+    if (claim.startDate > block.timestamp) revert IGachaLazyClaim.ClaimInactive();
+    if (claim.endDate > 0 && claim.endDate <= block.timestamp) revert IGachaLazyClaim.ClaimInactive();
+    if (msg.value < (claim.cost + MINT_FEE) * mintCount) revert IGachaLazyClaim.InvalidPayment();
+    uint32 amountAvailable = claim.totalMax - claim.total;
+    if (amountAvailable == 0 ) revert IGachaLazyClaim.ClaimSoldOut();
+
+    // calculate the amount to reserve and update totals
+    uint32 amountToReserve = uint32(Math.min(mintCount, amountAvailable));
+    claim.total += amountToReserve;
+    _reservedMintsPerWallet[creatorContractAddress][instanceId][msg.sender] += amountToReserve;
+
+    // Refund any overpayment
+    if (msg.value > (claim.cost + MINT_FEE) * amountToReserve){
+      uint256 refundAmount = msg.value - (claim.cost + MINT_FEE) * amountToReserve;
+      _sendFunds(payable(msg.sender), refundAmount);
+    }
+    emit GachaClaimMintReserved(creatorContractAddress, instanceId, msg.sender, amountToReserve);
+  }
+
+  /**
+   * See {IGachaLazyClaim-deliverMints}.
+   */
+  function deliverMints(IGachaLazyClaim.Mint[] calldata mints) external override {
+    _validateSigner();
+    _mint(mints);
+  }
+
+  function _mint(Mint[] calldata mints) private {
+    for (uint256 i; i < mints.length; ) {
+      Mint calldata mintData = mints[i];
+      _mintClaim(mintData);
```

Comment:
Combine deliverMints, _mint and _mintClaim.  No reason to separate it all

Reply:
You are looping twice for no reason, causing a lot of gas usage, because you've split out the functions